File: Explore/Learn/Introduction to Data Structure/Queue and Stack/Daily Temperatures/DailyTemp.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        int len = T.length;
        int[] result = new int[len];
        if (len == 1) return result;
        
        Stack<Integer> stack = new Stack<>();       
        for (int i=0; i<len; i++) {
            while (!stack.isEmpty() && T[stack.peek()] < T[i]) {
                int idx = stack.pop();
                result[idx] = i-idx;
            }
            stack.push(i);
        }
        
        return result;
    }
}

File: Explore/Learn/Introduction to Data Structure/Queue and Stack/Daily Temperatures/DailyTemp.kt
class Solution {
    fun dailyTemperatures(T: IntArray): IntArray {
        val result = IntArray(T.size)
        val stack = ArrayDeque<Int>()
        
        T.forEachIndexed {
            index, value ->
                while (!stack.isEmpty() && value > T[stack.peek()]) {
                    val top = stack.pop()
                    result[top] = index - top
                }
                stack.push(index)
        }
        
        return result
    }
}

